crates/api/src/local_user.rs

```diff
@@ -1072,36 +1137,43 @@ impl Perform for CreatePrivateMessage {
       Err(_e) => return Err(ApiError::err("couldnt_create_private_message").into()),
     };
 
-    updated_private_message.send_create(&user, context).await?;
-
-    // Send notifications to the recipient
-    let recipient_id = data.recipient_id;
-    let recipient_user =
-      blocking(context.pool(), move |conn| User_::read(conn, recipient_id)).await??;
-    if recipient_user.send_notifications_to_email {
-      send_email_to_user(
-        recipient_user,
-        "Private Message from",
-        "Private Message",
-        &content_slurs_removed,
-      );
-    }
+    updated_private_message
+      .send_create(&local_user_view.person, context)
+      .await?;
 
-    let message = blocking(context.pool(), move |conn| {
+    let private_message_view = blocking(context.pool(), move |conn| {
       PrivateMessageView::read(conn, inserted_private_message.id)
     })
     .await??;
 
     let res = PrivateMessageResponse {
-      private_message_view: message,
+      private_message_view,
     };
 
-    context.chat_server().do_send(SendUserRoomMessage {
-      op: UserOperation::CreatePrivateMessage,
-      response: res.clone(),
-      recipient_id,
-      websocket_id,
-    });
+    // Send notifications to the local recipient, if one exists
+    let recipient_id = data.recipient_id;
+    if let Ok(local_recipient) = blocking(context.pool(), move |conn| {
+      LocalUserView::read_person(conn, recipient_id)
+    })
+    .await?
+    {
+      if local_recipient.local_user.send_notifications_to_email {
+        send_email_to_user(
+          &local_recipient,
+          "Private Message from",
+          "Private Message",
+          &content_slurs_removed,
+        );
+      }
```

Comment:
Without this check, it crashed lemmy a few times. Anyways now for private messages, we check to make sure that user has a localuser row first.

Reply:
You mean the `if local_recipient.local_user.send_notifications_to_email`? I think it would be easier to move that check inside `send_email_to_user()`.

Reply:
Or even move this whole notification block into a helper method, cause you copy pasted the same code a couple times.